stamina-testkit/src/main/scala/stamina/testkit/StaminaTestKit.scala

```diff
@@ -0,0 +1,85 @@
+package stamina
+package testkit
+
+import scala.util._
+import scala.annotation.tailrec
+
+trait StaminaTestKit { self: org.scalatest.WordSpecLike ⇒
+
+  val defaultSampleId = "default"
+  case class PersistableSample(sampleId: String, persistable: AnyRef, description: Option[String]) {
+    override def toString = persistable.getClass.getSimpleName + description.map(" " + _).getOrElse("")
+  }
+
+  def sample(persistable: AnyRef) = new PersistableSample(defaultSampleId, persistable, None)
+  def sample(sampleId: String, persistable: AnyRef) = new PersistableSample(sampleId, persistable, Some(sampleId))
+  def sample(sampleId: String, persistable: AnyRef, description: String) = new PersistableSample(sampleId, persistable, Some(description))
+
+  implicit class TestablePersisters(persisters: Persisters) extends org.scalatest.Matchers {
+    def generateTestsFor(samples: PersistableSample*): Unit = {
+      samples.foreach { sample ⇒
+        performRoundtrip(sample)
+        deserializeStoredVersions(sample)
+      }
+      // Here we could verify test coverage
+    }
+
+    private def performRoundtrip(sample: PersistableSample) = {
+      s"persist and unpersist $sample" in {
+        persisters.canPersist(sample.persistable) should be(true)
+        persisters.unpersist(persisters.persist(sample.persistable)) should equal(sample.persistable)
+      }
+    }
+
+    @tailrec
+    private def deserializeStoredVersions(sample: PersistableSample, fromVersion: Int = 1): Unit = {
```

Comment:
It's not specifically testing the migrations, I setlled for `generateStoredVersionsDeserializationTestsFor`